ethreceipts/ethreceipts.go

```diff
@@ -0,0 +1,197 @@
+package ethreceipts
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"sync/atomic"
+	"time"
+
+	"github.com/0xsequence/ethkit/ethmonitor"
+	"github.com/0xsequence/ethkit/ethrpc"
+	"github.com/0xsequence/ethkit/go-ethereum/common"
+	"github.com/0xsequence/ethkit/go-ethereum/core/types"
+	"github.com/rs/zerolog"
+)
+
+type Receipts struct {
+	options Options
+
+	log      zerolog.Logger
+	provider *ethrpc.Provider
+
+	mointor *ethmonitor.Monitor
+
+	ctx     context.Context
+	ctxStop context.CancelFunc
+	running int32
+}
+
+type Options struct {
+	// NumBlocksUntilTxnFinality is the number of blocks that have passed after a txn has
+	// been mined to consider it have reached *finality* (aka its there for good now).
+	NumBlocksUntilTxnFinality int
+
+	// WaitNumBlocksBeforeExhaustion is the number of blocks that have passed since a get receipt
+	// request is made until we error with ErrExhausted. The client may want to try to find the receipt
+	// again, or it might re-price a txn, or other.
+	WaitNumBlocksBeforeExhaustion int
+}
+
+var DefaultOptions = Options{
+	NumBlocksUntilTxnFinality:     12,
+	WaitNumBlocksBeforeExhaustion: 3,
+}
+
+type Event uint32
+
+// We might not need Unknown here, we can use Discarded instead.
+const (
+	Finalized Event = iota
```

Comment:
usually better for Unknown to be the default zero if you're going to use it. This will ensure that we know the value was set to something useful which is non-default value of zero